Content.Server/_NF/Security/ContrabandGunSystem.cs

```diff
@@ -0,0 +1,76 @@
+using Content.Server.Popups;
+using Content.Shared._NF.Contraband.Components;
+using Content.Server._NF.Security.Components;
+using Content.Shared.IdentityManagement;
+using Content.Shared.Interaction;
+using Content.Shared.Timing;
+using Content.Shared.Verbs;
+
+namespace Content.Server._NF.Security.Systems;
+
+/// <summary>
+/// This handles...
+/// </summary>
+public sealed class ContrabandGunSystem : EntitySystem
+{
+    [Dependency] private readonly UseDelaySystem _useDelay = default!;
+    [Dependency] private readonly PopupSystem _popupSystem = default!;
+
+    /// <inheritdoc/>
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<ContrabandGunComponent, AfterInteractEvent>(OnAfterInteract);
+        SubscribeLocalEvent<ContrabandGunComponent, GetVerbsEvent<UtilityVerb>>(OnUtilityVerb);
+    }
+
+    private void OnUtilityVerb(EntityUid uid, ContrabandGunComponent component, GetVerbsEvent<UtilityVerb> args)
+    {
+        if (!args.CanAccess || !args.CanInteract || args.Using == null)
+            return;
+
+        if (!TryComp(uid, out UseDelayComponent? useDelay) || _useDelay.IsDelayed((uid, useDelay)))
+            return;
```

Comment:
I know this is from PriceGunSystem, just making a note for both systems - should this really require a UseDelayComponent?

If not, should probably be as follows:
```if (TryComp(uid, out UseDelayComponent? useDelay) && _useDelay.IsDelayed((uid, useDelay)))```

Reply:
Less spam